docs/source/release_notes/release-notes.rst

```diff
@@ -56,6 +56,13 @@ develop
     *    - |devbreak|
          - IteratorUtils.forEach removed; it's not needed in a Java 8 codebase.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/2207>`__)
+           
+    *    - |improved|
+         - Timelock now creates client namespaces the first time they are requested, rather than requiring them to be specified in config.
+           This means that specifying a list of clients in Timelock configuration will no longer have any effect. Further, a new configuration property called ``max-number-of-clients`` has been introduced in ``TimeLockRuntimeConfiguration``. This can be used to limit the number of clients that will be created dynamically, since each distinct client has some memory, disk space, and CPU overhead.
```

Comment:
It's pretty cool how this doesn't involve a dev break (well, unless someone has a test that querying for another client spits out some kind of error).

This could result in a little confusion if people try to debug their timelock and accidentally have a typo when doing a curl, but that's fine.